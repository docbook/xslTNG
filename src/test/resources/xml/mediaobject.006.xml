<?xml version="1.0" encoding="UTF-8"?>
<article xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink"
    version="5.1">
    <info>
        <title>SVG Graphic Dimensions</title>
        <author>
            <personname><firstname>Frank</firstname><surname>Steimke</surname></personname>
            <email>fsteimke.hb@gmail.com</email>
        </author>
    </info>
    <para>Check scaling an svg image where Dimensions are given in svg file in Inches: <code>&lt;sgv
            width="24.09375in" height="9.40625in"></code>. The image is created by the
            <productname>MagicDraw</productname> UML Software.</para>
    <para><link xlink:href="https://github.com/docbook/xslTNG/issues/432">Issue 432</link> is that
        the generated HTML creates an <code>img</code> with wrong size:
            <code>style="width:144px;height:96px;"</code> (which is <code>1.5in x 1.0in</code>). </para>
    <para>Function <code>f:object-properties</code>  in <filename>objects.xsl</filename> uses the
        external function <code>ext:image-metadata</code>, which seems to fail when dimensions in an
        SVG file are given as value with units.</para>
    <informaltable colsep="1" rowsep="1" frame="all">
        <tgroup cols="4">
            <colspec colnum="1" colname="col1" colwidth="2*" align="left"/>
            <colspec colnum="2" colname="col2" colwidth="1*" align="right"/>
            <colspec colnum="3" colname="col3" colwidth="1*" align="right"/>
            <colspec colnum="4" colname="col4" colwidth="1*" align="right"/>
            <thead>
                <row>
                    <entry/>
                    <entry namest="col2" nameend="col3" align="center">Original Size</entry>
                    <entry>Scaled 25%</entry>
                </row>
                <row>
                    <entry/>
                    <entry>Inch</entry>
                    <entry>px</entry>
                    <entry>px</entry>
                </row>
            </thead>
            <tbody>
                <row>
                    <entry>Height</entry>
                    <entry>9,40625</entry>
                    <entry>903</entry>
                    <entry>226</entry>
                </row>
                <row>
                    <entry>Width</entry>
                    <entry>24,09375</entry>
                    <entry>2313</entry>
                    <entry>578</entry>
                </row>
            </tbody>
        </tgroup>
    </informaltable>
    <figure>
        <title>SVG Image scaled by 25%</title>
        <mediaobject>
            <imageobject>
                <imagedata fileref="../media/EvidenceRequest.svg" scale="25"/>
            </imageobject>
            <caption>
                <para>This is an SVG Image scaled by 25%. Expected Dimensions are 578px x 226
                    px</para>
            </caption>
        </mediaobject>
    </figure>
    <figure>
        <title>SVG Image not scaled</title>
        <mediaobject>
            <imageobject>
                <imagedata fileref="../media/duck-small.svg"/>
            </imageobject>
            <caption>
                <para>This is an unscaled SVG Image. Neither <code>width</code> nor 
                    <code>depth</code> Attributes are expected.</para>
            </caption>
        </mediaobject>
    </figure>
    <figure>
        <title>SVG Image scaled by 110%</title>
        <mediaobject>
            <imageobject>
                <imagedata fileref="../media/duck-small.svg" scale="110"/>
            </imageobject>
            <caption>
                <para>This is the same SVG Image scaled by 110%, expected Dimensions are 440px x 430px </para>
            </caption>
        </mediaobject>
    </figure>
</article>
