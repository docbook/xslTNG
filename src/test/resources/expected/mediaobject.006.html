<!DOCTYPE html><html xmlns="http://www.w3.org/1999/xhtml" class="no-js"><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><script>(function(H){H.className=H.className.replace(/\bno-js\b/,'js')})(document.documentElement)</script><title>SVG Graphic Dimensions</title><meta name="viewport" content="width=device-width, initial-scale=1.0"/><link href="https://purl.org/dc/elements/1.1/" rel="schema.dc"/><meta content="2011-04-22T17:02:00-06:00" name="dc.modified"/><meta content="DocBook xslTNG" name="generator"/><link href="./css/docbook.css" rel="stylesheet" media="screen"/></head><body class="home"><nav class="top"></nav><main><article class="article component"><header><h1>SVG Graphic Dimensions</h1><div class="author"><h3><span class="first-last personname"><span class="firstname">Frank</span> <span class="surname">Steimke</span></span></h3></div></header><p>Check scaling an svg image where Dimensions are given in svg file in Inches: <code>&lt;sgv
            width="24.09375in" height="9.40625in"&gt;</code>. The image is created by the
            <span class="productname">MagicDraw</span> UML Software.</p><p><a href="https://github.com/docbook/xslTNG/issues/432" class="link">Issue 432</a> is that
        the generated HTML creates an <code>img</code> with wrong size:
            <code>style="width:144px;height:96px;"</code> (which is <code>1.5in x 1.0in</code>). </p><p>Function <code>f:object-properties</code>  in <code class="filename">objects.xsl</code> uses the
        external function <code>ext:image-metadata</code>, which seems to fail when dimensions in an
        SVG file are given as value with units.</p><div id="R_informaltable1" class="informalobject informaltable"><table><colgroup><col style="width: 40%"/><col style="width: 20%"/><col style="width: 20%"/><col style="width: 20%"/></colgroup><thead><tr><th class="bleft btop colsep left rowsep"></th><th class="btop center colsep rowsep" colspan="2">Original Size</th><th class="btop colsep right rowsep">Scaled 25%</th></tr><tr><th class="bleft colsep left rowsep"></th><th class="colsep right rowsep">Inch</th><th class="colsep right rowsep">px</th><th class="colsep right rowsep">px</th></tr></thead><tbody><tr><td class="bleft btop colsep left rowsep">Height</td><td class="btop colsep right rowsep">9,40625</td><td class="btop colsep right rowsep">903</td><td class="btop colsep right rowsep">226</td></tr><tr><td class="bleft colsep left rowsep">Width</td><td class="colsep right rowsep">24,09375</td><td class="colsep right rowsep">2313</td><td class="colsep right rowsep">578</td></tr></tbody></table></div><div id="R_fig1" class="figure formalobject"><div class="mediaobject"><div class="media image"><span class="viewport-table"><span class="viewport-row"><span class="viewport-cell" style="vertical-align:middle;"><span><span class="viewport"><picture class="imageobject"><img src="media/EvidenceRequest.svg" style="width:578px;height:226px;"/></picture></span></span></span></span></span></div><div class="caption"><p>This is an SVG Image scaled by 25%. Expected Dimensions are 578px x 226
                    px</p></div></div><header><div class="title">Figure <span class="label">1</span><span class="sep">. </span>SVG Image scaled by 25%</div></header></div><div id="R_fig2" class="figure formalobject"><div class="mediaobject"><div class="media image"><picture class="imageobject"><img src="media/duck-small.svg"/></picture></div><div class="caption"><p>This is an unscaled SVG Image. Neither <code>width</code> nor 
                    <code>depth</code> Attributes are expected.</p></div></div><header><div class="title">Figure <span class="label">2</span><span class="sep">. </span>SVG Image not scaled</div></header></div><div id="R_fig3" class="figure formalobject"><div class="mediaobject"><div class="media image"><span class="viewport-table"><span class="viewport-row"><span class="viewport-cell" style="vertical-align:middle;"><span><span class="viewport"><picture class="imageobject"><img src="media/duck-small.svg" style="width:440px;height:430px;"/></picture></span></span></span></span></span></div><div class="caption"><p>This is the same SVG Image scaled by 110%, expected Dimensions are 440px x 430px </p></div></div><header><div class="title">Figure <span class="label">3</span><span class="sep">. </span>SVG Image scaled by 110%</div></header></div></article></main><nav class="bottom"></nav></body></html>